android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/SetupWalletFragment.java

```diff
@@ -57,12 +62,14 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         super.onViewCreated(view, savedInstanceState);
         Button setupCryptoButton = view.findViewById(R.id.btn_setup_crypto);
         setupCryptoButton.setOnClickListener(v -> {
+            WalletDataFilesInstaller.registerWalletDataFilesComponentOnDemand();
             checkOnBraveActivity(true, false);
             onNextPage.gotoOnboardingPage();
         });
 
         TextView restoreButton = view.findViewById(R.id.btn_restore);
         restoreButton.setOnClickListener(v -> {
+            WalletDataFilesInstaller.registerWalletDataFilesComponentOnDemand();
```

Comment:
I chose to start download after either `Get started`/`Restore` button click because, like you mentioned, if user will not create wallet - then component files will remain on disk.

What are the chances that user open Wallet screen, but dosn't do the initial setup then? - Probably they are not so high.

In any way both solutions - download immediately or on button click are better than it was when the component was downloaded right after app start. So I'll change it.

Reply:
changed at acc95adc4fddcf833676d6f9979e793c16ed70fa

Reply:
Yeah I think in most cases users continue to create or restore wallet, it will be more beneficial to get everything ready when they done.